build.sh

```diff
@@ -150,14 +150,11 @@ restore_optdata()
 
     if [ $__isMSBuildOnNETCoreSupported == 1 ]; then
         # Parse the optdata package versions out of msbuild so that we can pass them on to CMake
-        local DotNetCli="$__ProjectRoot/.dotnet/dotnet"
-        if [ ! -f $DotNetCli ]; then
-            source "$__ProjectRoot/init-dotnet.sh"
-            if [ $? != 0 ]; then
-                echo "Failed to install dotnet."
-                exit 1
-            fi
-        fi
+
+        # Init dotnet
+        source "${__ProjectRoot}/init-dotnet.sh"
+        local DotNetCli=${DOTNET_INSTALL_DIR}/dotnet
```

Comment:
Is there a reason we can't just call `dotnet.sh` instead in the two uses of `DotNetCli` below?

Reply:
We can, it requires one of the following to happen:

1. dotnet.sh will need to remove the echo and will not be allowed to log
2. dotnet.sh will need a verbosity flag to keep it from logging.

Reply:
Seems like arcade's install script also has output. This has to stay the same simply because of the fragile way we are expecting text output from the call.